libs/esl/ivrd.c

```diff
@@ -41,8 +41,12 @@
 static void handle_SIGCHLD(int sig)
 {
 	int status = 0;
+	int pid = 0;
+
+	do {
+		pid = waitpid(-1, &status, WNOHANG);
+	} while (pid > 0);
```

Comment:
Is this going to burn CPU? Do we need a small sleep?

Reply:
No, there is no CPU burning because this function is only called when SIGCHILD is found and with WNOHANG flag we prevent from blocking process, it will return current process status and if there is an error it will return immediatly this error (PID .

We already test on our platform this patch (3000 simultaneous call) and no CPU burning was found with our monitoring